tough/src/editor/targets.rs

```diff
@@ -300,7 +300,7 @@ impl TargetsEditor {
                     delegated_role.keyids.extend(keyids.clone());
                 }
             }
-            for delegated_role in self.new_roles.get_or_insert(Vec::new()).iter_mut() {
+            for delegated_role in &mut *self.new_roles.get_or_insert(Vec::new()) {
```

Comment:
I'm a little surprised at this one. Do you know what the reasoning behind it is?

Reply:
Looks like it is implicitly creating an iterator, the issue may be `get_or_insert` instead of `unwrap_or_default`. I'm not sure if that creates a cleaner loop or not.

Reply:
The actual warning from the existing code is:

```
warning: it is more concise to loop over references to containers instead of using explicit iteration methods
```

With this being given as the suggested change.

https://rust-lang.github.io/rust-clippy/master/index.html#/explicit_iter_loop

The `&mut *` part looks a bit confusing at first, but I believe this _should_ be correct of iterating over a mutable reference to each element. Please correct me if I am interpreting that wrong.